synthwave-theme.el

```diff
@@ -1,139 +1,400 @@
-
-(provide-theme 'synthwave)
-
-;; Local Variables:
-;; no-byte-compile: t
-;; End:
+;;; synthwave-theme.el --- 1980's Emacs color theme on dark background
+
+;; Copyright (C) 2003-2019 Free Software Foundation, Inc.
+
+;; Author:TroyFletcher
+;; URL: https://github.com/TroyFletcher/emacs-synthwave-theme
+;; Version: 201906...
+;; Keywords: color theme
+
+;; This theme was generated with vimco.el
+;;   You can get it from:
+;;   <https://github.com/DoMiNeLa10/vimco>
+;;
+;;   Absolutely legend, this guy ^^^^ -Troy
+;;
+;; This file is part of GNU Emacs.
+
+;; GNU Emacs is free software: you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; GNU Emacs is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with GNU Emacs.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; This high-contrast 1980's retro cyber theme uses bright colors on a
+;; dark background.
+;;
+;; To use it, put the following in your Emacs configuration file:
+;;
+;;   (load-theme 'synthwave t)
+;;
+;; Code remodeled to match leuven-theme.el style and extended to more modes
+;;
+;; Requirements: Emacs 24+.
+;;
+
+;;; Code:
+
+;;; Options.
+;;; Lifted directly from leuven-theme
+(defgroup synthwave nil
+  "Synthwave theme options.
+The theme has to be reloaded after changing anything in this group."
+  :group 'faces)
+
+(defcustom synthwave-scale-outline-headlines t
+  "Scale `outline' (and `org') level-1 headlines.
+This can be nil for unscaled, t for using the theme default, or a scaling number."
+  :type '(choice
+          (const :tag "Unscaled" nil)
+          (const :tag "Default provided by theme" t)
+          (number :tag "Set scaling"))
+  :group 'synthwave)
+
+(defcustom synthwave-scale-org-agenda-structure t
+  "Scale Org agenda structure lines, like dates.
+This can be nil for unscaled, t for using the theme default, or a scaling number."
+  :type '(choice
+          (const :tag "Unscaled" nil)
+          (const :tag "Default provided by theme" t)
+          (number :tag "Set scaling"))
+  :group 'synthwave)
+
+(defun synthwave-scale-font (control default-height)
+  "Function for splicing optional font heights into face descriptions.
+CONTROL can be a number, nil, or t.  When t, use DEFAULT-HEIGHT."
+  (cond
+   ((numberp control) (list :height control))
+   ((eq t control) (list :height default-height))
+   (t nil)))
+
+;;; Theme Faces
+(deftheme synthwave
+  "Dark retro 1980's theme with bold colors.")
+
+(let* (
+       (class '((class color) (min-colors 89)))
+       ;; Synthwave colors
+       (near-black "#100c0c")
```

Comment:
I want a lighter background to better contrast the full black font-lock-comment-face. Original is "#181615" It can be assigned as near-black or at the default-background designation.